File: api/src/models/user.model.ts
import { DataTypes } from "sequelize"
import { sequelize } from "../config/db"

export const User = sequelize.define("user", {
	uid: {
		type: DataTypes.STRING,
	},
	name: {
		type: DataTypes.STRING,
		allowNull: false        
	},
	email: {
		type: DataTypes.STRING,
		allowNull: false
	},
	password: {
		type: DataTypes.STRING,
		allowNull: false
	},
	phoneNumber: {
		type: DataTypes.STRING,
	},
	image: {
		type: DataTypes.STRING,
	},
	balance: {
		type: DataTypes.STRING,
	},
	points: {
		type: DataTypes.STRING,
	},
})

// export class User extends Model {
//     declare id :number;
//     declare name: string;
//     declare email: string;
//     declare password: string;
//     declare phoneNumber: string;
//     declare image: string; // Change the type to Buffer | null
//     declare balance: number;
//     declare points: number;
// };

// User.init(
//     {
//         id: {
//             type: new DataTypes.INTEGER,
//             primaryKey: true,
//             autoIncrement: true,
//         },
//         // uid: {
//         //     type: new DataTypes.STRING,
//         //     allowNull: true
//         // },
//         name: {
//             type: new DataTypes.STRING,
//             allowNull: false
//         },
//         email: {
//             type: new DataTypes.STRING,
//             allowNull: false
//         },
//         password: {
//             type: new DataTypes.STRING,
//             allowNull: false
//         },
//         phoneNumber: {
//             type: new DataTypes.STRING,
//             allowNull: false
//         },
//         // image: {
//         //     type: new DataTypes.STRING,
//         //     allowNull: true,
//         // },
//         balance: {
//             type: new DataTypes.INTEGER,
//             allowNull: true
//         },
//         points: {
//             type: new DataTypes.INTEGER,
//             allowNull: true
//         },
//         createdAt: {
//             type: new DataTypes.DATE,
//             allowNull: true
//         },
//         updatedAt: {
//             type: new DataTypes.DATE,
//             allowNull: true
//         }
//     },
//     {
//         tableName: 'user',
//         sequelize,
//     },
// )

File: lib/models/user_model.dart

class User {
  final String uid;
  final String name;
  final String email;
  final String? phoneNumber;
  final String? image;
  final double? balance;
  final int? points;
  final DateTime createdAt;
  final DateTime updatedAt;
  final List<String>? reward;

  User({
    required this.uid,
    required this.name,
    required this.email,
    this.phoneNumber,
    this.image,
    this.balance,
    this.points,
    required this.createdAt,
    required this.updatedAt,
    this.reward,
  });

  factory User.fromJson(Map<String, dynamic> json) {
    // Parse the 'reward' field as a List<String>
    List<String> rewardList = json['reward'] != null ? List<String>.from(json['reward']) : [];

    return User(
      uid: json['uid'],
      name: json['name'],
      email: json['email'],
      phoneNumber: json['phoneNumber'],
      image: json['image'],
      balance: json['balance'] != null ? json['balance'].toDouble() : null,
      points: json['points'],
      createdAt: DateTime.parse(json['createdAt']),
      updatedAt: DateTime.parse(json['updatedAt']),
      reward: rewardList,
    );
  }
}
